packages/core/useFetch/index.ts

```diff
@@ -0,0 +1,161 @@
+import { Ref, ref, unref, watch, isRef, computed, ComputedRef } from 'vue-demi'
+import { Fn, MaybeRef } from '@vueuse/shared'
+
+interface UseFetchReturn {
+  /**
+   * Indicates if the fetch request has finished
+   */
+  isFinished: Ref<boolean>
+
+  /**
+   * The status of the fetch request
+   */
+  status: Ref<number | null>
+
+  /**
+   * The raw response of the fetch request
+   */
+  response: Ref<Response | null>
+
+  /**
+   * Any fetch errors that may have occured
+   */
+  error: Ref<any>
+
+  /**
+   * The fetch response body, may either be JSON or text
+   */
+  data: Ref<object | string | null>
+
+  /**
+   * Indicates if the request is currently being fetched.
+   */
+  isFetching: Ref<boolean>
+
+  /**
+   * Indicates if the fetch request is able to be aborted
+   */
+  canAbort: ComputedRef<boolean>
+
+  /**
+   * Abort the fetch request
+   */
+  abort: Fn
+
+  /**
+   * Manually call the fetch
+   */
+  execute: Fn
+}
+
+interface UseFetchOptions {
+  /**
+   * Will automatically run fetch when `useFetch` is used
+   *
+   * @default true
+   */
+  autoFetch?: boolean
+
+  /**
+   * Will automatically refetch when the URL is changed if the url is a ref
+   *
+   * @default false
+   */
+  autoRefetch?: MaybeRef<boolean>
+}
+
+export function useFetch(url: MaybeRef<string>): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, useFetchOptions: UseFetchOptions): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, options: RequestInit, useFetchOptions?: UseFetchOptions): UseFetchReturn
+
+export function useFetch(url: MaybeRef<string>, ...args: any[]): UseFetchReturn {
+  const supportsAbort = typeof AbortController === 'function'
+
+  let fetchOptions: RequestInit = {}
+  let options: UseFetchOptions = { autoFetch: true, autoRefetch: false }
+
+  if (args[0]) {
```

Comment:
Let's check the overload with `args.length` instead of relying on their properties.